app/services/youtube_service.rb

```diff
@@ -0,0 +1,26 @@
+class YoutubeService
+  def initialize(episode)
+    @episode = episode
+    @video = Yt::Video.new id: episode.video
+    Yt.configuration.api_key = ENV['TEST_CHANNELKEY']
+  end
+
+  def broadcast_start_date
+    @video.actual_start_time
+  end
+
+  def broadcast_end_date
+    @video.actual_end_time
+  end
+
+  def youtube_status
+    @video.live_broadcast_content == 'none' ? 'over' : @video.live_broadcast_content
+  end
+
+  def update
+    @episode.broadcast_begin = broadcast_start_date
```

Comment:
```ruby
@episode.update(broadcast_begin: @video.actual_start_time, ...)
```
Btw you can just include this `update` to the model.

Reply:
I guess adding it to the model won't do, because model doesn't validate youtube link when episode is created, if I understand your point correctly. Our workflow includes creating episode without any youtube link (as a draft).

Reply:
> I guess adding it to the model won't do, because model doesn't validate youtube link when episode is created, if I understand your point correctly.
https://guides.rubyonrails.org/v3.2/active_record_validations_callbacks.html
You can validate it in the model as I said before.

Reply:
Even if we don't add this logic to model - 

> ```ruby
> @episode.update(broadcast_begin: @video.actual_start_time, ...)
> ```

Change it to this at least